android/gltfio-android/src/main/cpp/FilamentAsset.cpp

```diff
@@ -65,6 +65,24 @@ Java_com_google_android_filament_gltfio_FilamentAsset_nGetEntities(JNIEnv* env,
     env->ReleaseIntArrayElements(result, (jint*) entities, 0);
 }
 
+extern "C" JNIEXPORT jint JNICALL
+Java_com_google_android_filament_gltfio_FilamentAsset_nGetMaterialInstanceCount(JNIEnv*, jclass,
+        jlong nativeAsset) {
+    FilamentAsset* asset = (FilamentAsset*) nativeAsset;
+    return asset->getMaterialInstanceCount();
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_google_android_filament_gltfio_FilamentAsset_nGetMaterialInstances(JNIEnv* env, jclass,
+        jlong nativeAsset, jlongArray result) {
+    FilamentAsset* asset = (FilamentAsset*) nativeAsset;
+    jsize available = env->GetArrayLength(result);
+    jsize count = std::min(available, (jsize) asset->getMaterialInstanceCount());
+    MaterialInstance** mats = (MaterialInstance**) env->GetLongArrayElements(result, nullptr);
+    std::copy_n(asset->getMaterialInstances(), count, mats);
```

Comment:
Will this work correctly on a 32 bit machine?

Reply:
Wow excellent catch, this code was wrong (fine on device, crash on emulator).  Fixed.